### PR TITLE
Increasing Line Coverage

### PR DESCRIPTION
**Pull request by caarlos0**
Refs #97

### CODE REVIEW

**Comment by karato** [comment]
Thanks, someone will review your pull request soon

**Comment by karato** [comment]
@darkled review it

File: src/test/java/com/nerodesk/aws/AwsDocTest.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+     * @throws Exception in case of error.
+     */
+    @Test
+    public void exists() throws Exception {
+        final Ocket ocket = Mockito.mock(Ocket.class);
+        final String label = "document-label";
+        Mockito.when(this.bucket.ocket(label)).thenReturn(ocket);
+        Mockito.when(ocket.exists()).thenReturn(true);
+        MatcherAssert.assertThat(
+            new AwsDoc(this.bucket, "non-existant-document").exists(),
```
Comment (asaen) [inline src/test/java/com/nerodesk/aws/AwsDocTest.java]: looks like a typo, should be "non-exist**e**nt-document"

File: src/test/java/com/nerodesk/aws/AwsDocTest.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+     */
+    @Test
+    public void deletes() throws Exception {
+        final String label = "document-to-delete";
+        new AwsDoc(this.bucket, label).delete();
+        Mockito.verify(this.bucket).remove(label);
+    }
+
+    /**
+     * AwsDoc can read itself to a OutputStream.
```
Comment (asaen) [inline src/test/java/com/nerodesk/aws/AwsDocTest.java]: read from an OutputStream ?

**Comment by asaen** [comment]
@caarlos0 looks nice, just a couple minor comments above.

File: src/test/java/com/nerodesk/aws/AwsDocTest.java (hunk 1 above)
Comment (caarlos0) [inline src/test/java/com/nerodesk/aws/AwsDocTest.java]: good catch

**Comment by caarlos0** [comment]
Had some conflicts with @krzyk PR, merged and fixed all.

**Comment by asaen** [comment]
@caarlos0 great, thanks!

**Comment by asaen** [comment]
@rultor merge

**Comment by yegor256** [comment]
@caarlos0 all these tests would look much more compact and readable if you would use classes from `com.jcabi.s3.mock` package, see http://www.yegor256.com/2014/09/23/built-in-fake-objects.html

**Comment by asaen** [comment]
@caarlos0 there are some new conflicts as well..

**Comment by caarlos0** [comment]
@yegor256 there tests for this using Mockito already, I'll add a puzzle to fix that.

**Comment by caarlos0** [comment]
@yegor256 all done.

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by asaen** [comment]
@caarlos0 it seems there are still some merge conflicts...

**Comment by caarlos0** [comment]
@darkled nope, there are merge conflicts again.

**Comment by caarlos0** [comment]
Build failing because of #159

**Comment by asaen** [comment]
@karato we're waiting for #159 here

**Comment by karato** [comment]
> @karato we're waiting for #159 here

@darkled right, let's wait for #159

**Comment by karato** [comment]
we waited for #159, it is closed already

**Comment by asaen** [comment]
@rultor let's try to merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by asaen** [comment]
@caarlos0 please fix the puzzle description in `package-info.java`

```
ERROR: "puzzle src/test/java/com/nerodesk/om/aws/package-info.java:37-38 has a very short description of just 19 words while a minimum of 20 is required" /var/lib/gems/1.9.1/gems/pdd-0.14/lib/pdd.rb:149:in `rules': 1 errors, see log above (PDD::Error)
```

**Comment by caarlos0** [comment]
@darkled fixed.

**Comment by asaen** [comment]
@rultor please try to merge once more

**Comment by caarlos0** [comment]
@yegor256 can you check this please?

**Comment by yegor256** [comment]
@rultor merge

**Comment by karato** [comment]
@darkled Thanks a lot, I just topped your account for **24 mins**, transaction ID `54595755`, total time was 28 hours and 57 mins

extra minutes for review comments (c=9)

+24 added to your rating, at the moment it is: [+2243](http://www.netbout.com/b/35091?open=rating)

**Comment by karato** [comment]
@rultor deploy

**Pull request merged**